samples/apps/copilot-chat-app/webapi/Controllers/ChatHistoryController.cs

```diff
@@ -0,0 +1,193 @@
+    }
+
+    /// <summary>
+    /// Get all chat sessions associated with a user. Return an empty list if no chats are found.
+    /// </summary>
+    /// <param name="userId">The user id.</param>
+    [Authorize]
+    [HttpGet]
+    [Route("chat/getAllChats/{userId:regex([[0-9]]+\\.[[0-9]]+)}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetAllChatsAsync(string userId)
+    {
+        var chats = await this._chatSessionRepository.FindByUserIdAsync(userId);
+        if (chats == null)
+        {
+            // Return an empty list if no chats are found
+            return this.Ok(new List<ChatSession>());
+        }
+
+        return this.Ok(chats);
+    }
+
+    /// <summary>
+    /// Get all chat messages for a chat session.
+    /// The list will be ordered with the first entry being the most recent message.
+    /// </summary>
+    /// <param name="chatId">The chat id.</param>
+    /// <param name="startIdx">The start index at which the first message will be returned.</param>
+    /// <param name="count">The number of messages to return. -1 will return all messages starting from startIdx.</param>
+    /// [Authorize]
+    [HttpGet]
+    [Route("chat/getChatMessages/{chatId:guid}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetChatMessagesAsync(
+        Guid chatId,
+        [FromQuery] int startIdx = 0,
+        [FromQuery] int count = -1)
+    {
+        var chatMessages = await this._chatMessageRepository.FindByChatIdAsync(chatId.ToString());
+        if (chatMessages == null)
+        {
+            return this.NotFound($"No messages found for chat of id {chatId}.");
+        }
+
+        if (startIdx >= chatMessages.Count())
+        {
+            return this.BadRequest($"Start index {startIdx} is out of range.");
+        }
+        else if (startIdx + count > chatMessages.Count() || count == -1)
+        {
+            count = chatMessages.Count() - startIdx;
+        }
+
+        chatMessages = chatMessages.OrderByDescending(m => m.Timestamp).Skip(startIdx).Take(count);
+        return this.Ok(chatMessages);
+    }
+
+    /// <summary>
+    /// Edit a chat session.
+    /// </summary>
+    /// <param name="chatParameters">Object that contains the parameters to edit the chat.</param>
+    [Authorize]
+    [HttpPost]
+    [Route("chat/edit")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> EditChatAsync([FromBody] ChatSession chatParameters)
+    {
+        var chatId = chatParameters.Id;
+        Console.WriteLine(chatId);
```

Comment:
Feel like this can be removed. Besides, to my knowledge, we should use ILogger for logs.

Reply:
Do we need the console log?